lnwallet/wallet.go

```diff
@@ -770,13 +771,36 @@ func (l *LightningWallet) handleFundingReserveRequest(req *InitFundingReserveMsg
 	// funder in the attached request to provision the inputs/outputs
 	// that'll ultimately be used to construct the funding transaction.
 	if !ok {
+		var err error
+		var numAnchorChans int
+
+		// Get the number of anchor channels to determine if there is a reserved
+		// value that must be respected when funding up to the maximum amount.
+		// Since private channels (most likely) won't be used for routing other
+		// than the last hop, they bear a smaller risk that we must force close
+		// them in order to resolve a HTLC up/downstream. Hence we exclude them
+		// from the count of anchor channels.
+		isPublic := req.Flags&lnwire.FFAnnounceChannel != 0
```

Comment:
It's true that we don't count private channels towards those that require a reserve. But we shouldn't include the `isPublic` in the `if` here, otherwise we'll run into an error if we're trying to open a private channel but already have a bunch of public channels that require the reserve.

Reply:
I see.
I argued to let `fundmax` work with private channels without reserving sats here: https://github.com/lightningnetwork/lnd/pull/4029#discussion_r883931794

Is there another way we can address this?

Reply:
Could we check here if we already have a public channel with that peer or would that be too slow? Or could we do this check maybe further upstream in the rpcserver? These are just vague shots :-)